numpy/core/src/npysort/selection.c.src

```diff
@@ -207,8 +216,39 @@ static NPY_INLINE void
         SWAP(SORTEE(*ll), SORTEE(*hh));
     }
 }
-
-
+/* partition for rivest-floyd algorithm around the v[IDX(*ll)] pivot.
+Return the index of the pivot
+*/
+npy_intp @name@rf_partition_@suff@(@type@ *v,
+#if @arg@
+                                 npy_intp *tosort,
+#endif
+                                 npy_intp low,
+                                 npy_intp high)
+{
+	npy_intp i = low;
+	npy_intp j = high;
+    @type@ pivot = v[IDX(low)];
+    //if pivots stays at low, it is going to be swapped to high
+    npy_bool pivot_low = 0;
+    //ensure v[IDX(low)] >= v[IDX(high)]
+    if (@TYPE@_LT(pivot, v[IDX(high)])) {
+        SWAP(SORTEE(low), SORTEE(high));
+        pivot_low = 1;
+        };
+    while (i < j) {
+        SWAP(SORTEE(i), SORTEE(j));
+        do i++; while (@TYPE@_LT(v[IDX(i)], pivot));
+        do j--; while (@TYPE@_LT(pivot, v[IDX(j)]));
+    };
+    //now put pivot in the jth place
+    if (pivot_low) SWAP(SORTEE(low), SORTEE(j))
+    else {
+        j++;
+        SWAP(SORTEE(high), SORTEE(j));
+    };
```

Comment:
Semicolons and braces are a little odd here,

```suggestion
    if (@TYPE@_LT(pivot, v[IDX(high)])) {
        SWAP(SORTEE(low), SORTEE(high));
        pivot_low = 1;
    }
    while (i < j) {
        SWAP(SORTEE(i), SORTEE(j));
        do i++; while (@TYPE@_LT(v[IDX(i)], pivot));
        do j--; while (@TYPE@_LT(pivot, v[IDX(j)]));
    }
    // now put pivot in the jth place
    if (pivot_low) {
        SWAP(SORTEE(low), SORTEE(j));
    }
    else {
        j++;
        SWAP(SORTEE(high), SORTEE(j));
    }
```

Reply:
Thank you for the review! I've implemented all suggestions and added another test for fuller code coverage.